src/main/java/de/tum/in/www1/artemis/service/connectors/iris/dto/IrisMessageResponseDTO.java

```diff
@@ -1,6 +1,27 @@
 package de.tum.in.www1.artemis.service.connectors.iris.dto;
 
-import de.tum.in.www1.artemis.domain.iris.IrisMessage;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
 
-public record IrisMessageResponseDTO(String usedModel, IrisMessage message) {
+import java.time.ZonedDateTime;
+
+public record IrisMessageResponseDTO(String usedModel, ZonedDateTime sentAt, JsonNode content) {
+    
+    /**
+     * Create a new IrisMessageResponseDTO. Jackson uses this constructor to create the object.
+     * This is necessary because Jackson was not throwing an exception when the response from Iris did not contain
+     * the expected fields, which resulted in a NullPointerException when trying to access the fields.
+     * Not sure if this is a bug in Jackson or if it is intended behavior, either way this is a workaround.
+     */
```

Comment:
Could you maybe mention your findings in the next dev meeting? Especially DTOs should be validated by Jackson, so maybe we can find a better solution, as your way kinda defeats the simpleness of records.